### PR TITLE
increase modularity of infer_cp_features

### LINKED ISSUE #81: Increase modularity of infer_cp_features
Currently, [`pycytominer.utils.features.infer_cp_features()`](https://github.com/cytomining/pycytominer/blob/master/pycytominer/cyto_utils/features.py) has `Cells`, `Nuclei`, and `Cytoplasm` hardcoded. 

We should extend this functionality to deal with more compartments.

### PR DESCRIPTION
**Pull request by hillsbury**
closes #81

### CODE REVIEW

**Comment by hillsbury** [comment]
@gwaygenomics i changed up the tests and should be ready for review now

File: pycytominer/cyto_utils/features.py [hunk 1]
```diff
@@ -66,19 +66,17 @@ def label_compartment(cp_features, compartment, metadata_cols):
...
-            x.startswith("Cells_")
-            | x.startswith("Nuclei_")
-            | x.startswith("Cytoplasm_")
-        )
-    ]
+    features = []
+    for col in population_df.columns.tolist():
+        for ind, val in enumerate(compartments):
+            if col.startswith(compartments[ind].title()):
+                features.append(col)
```
Comment (gwaygenomics) [inline pycytominer/cyto_utils/features.py:79]: Looks great @hillsbury - I've proposed a small change below to make the code more [pythonic](https://docs.python-guide.org/writing/style/).

```suggestion
        if any([col.startswith(x.title()) for x in compartments]):
            features.append(col)
```

I didn't test this, but I think it should work

File: pycytominer/tests/test_cyto_utils/test_feature_infer.py [hunk 2]
```diff
@@ -54,3 +54,14 @@ def test_metadata_feature_infer():
     expected = ["Metadata_Something_Something"]
 
     assert features == expected
+
+
+def test_feature_infer_compartments():
```
Comment (gwaygenomics) [inline pycytominer/tests/test_cyto_utils/test_feature_infer.py:59]: 🎉

**Review by gwaygenomics — CHANGES_REQUESTED** [review]
Looks great @hillsbury - I proposed one small suggestion. Once committed and tests pass, I will approve the PR and we'll get this change merged into master 🎊

**Comment by gwaygenomics** [comment]
Looks great @hillsbury - I will go ahead and approve+merge

**Review by gwaygenomics — APPROVED** [review]

**Pull request merged**